app-backend/db/src/test/scala/com/azavea/rf/database/CampaignDaoSpec.scala

```diff
@@ -511,4 +511,321 @@ class CampaignDaoSpec
+              TaskDao.updateTask(
+                f.id,
+                Task.TaskFeatureCreate(
+                  f.properties.toCreate
+                    .copy(
+                      status = TaskStatus.Validated,
+                      parentTaskId = Some(tasks.features.head.id),
+                      taskType = Some(TaskType.Review)
+                    ),
+                  f.geometry,
+                  "Feature"
+                ),
+                parent
+              )
+            }
+            randomTask <- CampaignDao
+              .randomReviewTask(insertedCampaign.id, parent)
+          } yield randomTask
+
+          val randomTask = copyIO.transact(xa).unsafeRunSync
+
+          assert(
+            randomTask.isEmpty,
+            "When all tasks are validated no tasks should be returned"
+          )
+
+          true
+        }
+      )
+    }
+  }
+
+  test("a task is returned when none have been validated") {
+    check {
+      forAll(
+        (
+            userCreate1: User.Create,
+            userCreate2: User.Create,
+            userCreates: List[User.Create],
+            userCreate: User.Create,
+            campaignCreate: Campaign.Create,
+            annotationProjectCreate: AnnotationProject.Create,
+            taskFeaturesCreate: Task.TaskFeatureCollectionCreate
+        ) => {
+          val copyIO = for {
+            parent <- UserDao.create(userCreate)
+            children <- userCreate1 :: userCreate2 :: userCreates traverse {
+              u =>
+                UserDao.create(u)
+            }
+            insertedCampaign <- CampaignDao
+              .insertCampaign(
+                campaignCreate.copy(parentCampaignId = None),
+                parent
+              )
+            insertedProject <- AnnotationProjectDao
+              .insert(
+                annotationProjectCreate.copy(
+                  campaignId = Some(insertedCampaign.id),
+                  status = AnnotationProjectStatus.Waiting
+                ),
+                parent
+              )
+            _ <- AnnotationProjectDao.update(
+              insertedProject.toProject
+                .copy(status = AnnotationProjectStatus.Ready),
+              insertedProject.id
+            )
+            campaignCopies <- children traverse { child =>
+              CampaignDao.copyCampaign(insertedCampaign.id, child)
+            }
+            projectCopies <- (campaignCopies traverse { c =>
+              AnnotationProjectDao.listByCampaign(c.id)
+            }) map (_.flatten)
+            projectCopiesWithRelated <- projectCopies traverse { pc =>
+              AnnotationProjectDao.getWithRelatedById(pc.id)
+            } map (_.flatten)
+            tasks <- TaskDao.insertTasks(
+              fixupTaskFeaturesCollection(
+                taskFeaturesCreate,
+                projectCopiesWithRelated.head
+              ),
+              parent
+            )
+            _ <- TaskDao.updateTask(
```

Comment:
Similar question to the first one -- does there need to be a `Label` task for the parent/child thinig?